### PR TITLE
Functional programming

### PR DESCRIPTION
**Pull request by Jamim**
Hosted for testing [here](http://aur.im/js-2015/Functional programming/).

### CODE REVIEW

File: Functional programming/Problem 1/partial_application.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+function F() {
+	"use strict";
+
+	var toArray = Array.prototype.slice;
+	var argsCount = arguments.length - 1;
+	var args = toArray.call(arguments, 0, argsCount);
+	var g = arguments[argsCount];
```
Comment (AntonNekrasov) [inline Functional programming/Problem 1/partial_application.js:7]: Supposition, that the last argument should be a function is a bad practice.  The more elegant way is to make it an implicit parameter: Function.prototype.partial = function () { ....

File: Functional programming/Problem 1/partial_application.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+function F() {
```
Comment (AntonNekrasov) [inline Functional programming/Problem 1/partial_application.js:1]: Using more intuitive names will increase your code [readability](http://javarevisited.blogspot.com/2014/10/10-java-best-practices-to-name-variables-methods-classes-packages.html).

File: Functional programming/Problem 2/currying.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+function curry(f) {
+	"use strict";
+
+	var len = f.length;
+	if (len == 0)
```
Comment (AntonNekrasov) [inline Functional programming/Problem 2/currying.js:5]: Use '===' to compare with '0'

File: Functional programming/Problem 2/currying.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+function curry(f) {
```
Comment (AntonNekrasov) [inline Functional programming/Problem 2/currying.js:1]: Again: making function an implicit parameter would ease your live

File: Functional programming/Problem 1/partial_application.js (hunk 2 above)
Comment (Jamim) [inline Functional programming/Problem 1/partial_application.js:1]: @AntonNekrasov
It's clear, but under the terms of the problem 1 I need to implement function _F_.

> **Problem 1: Partial Application**
> Implement function _F_ that allows to do…

File: Functional programming/Problem 5/map.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+function map(f, array) {
+	"use strict";
+
+	var new_array = [];
```
Comment (AntonNekrasov) [inline Functional programming/Problem 5/map.js:4]: naming conventions violation

File: Functional programming/Problem 11/memoization.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+function memoization(f) {
```
Comment (AntonNekrasov) [inline Functional programming/Problem 11/memoization.js:1]: How do you check, if you return exactly needed function?

**Pull request closed without merging**